src/nlp/arrow_writer.py

```diff
@@ -100,10 +100,29 @@ def _write_array_on_file(self, pa_array):
         self._num_bytes += pa_array.nbytes
         self.pa_writer.write_batch(pa_batch)
 
+    def _write_including_extension(self, arrays, schemas):
+        rows = map(lambda x: list(x.values()), self.current_rows)
+        for array, schema in zip(rows, schemas):
+            sorted_array = sorted(
+                [(x, str(type(x))) for x in array],
+                key=lambda x : x[1])
+            sorted_array = [x[0] for x in sorted_array]
+            sorted_schema = sorted(
+                [(x, str(x.type.__class__)) for x in schema],
+                key=lambda x : x[1])
+            sorted_schema = [x[0] for x in sorted_schema]
+            pa_batch = pa.RecordBatch.from_arrays(sorted_array, schema=schema)
+            self._num_bytes += pa_batch.nbytes
+            self.pa_writer.write_batch(pa_batch)
```

Comment:
I'm not sure I understand this part, could you provide more details ?

Reply:
ofc! I wrote this function here because originally in the write_on_file method, it would cast the self.current_rows to a pa.array which would throw an error when there was a column that was of the pa.ExtensionType type. So, to circumvent this I converted self.current rows into a list of individual rows where each row would be one batch (line 104). The only problem then became that the order of the fields in the schema and those of the values in the array (line 105) would not be ordered correctly if there would be more than one pa.ExtensionType types present in the schema. So as a temporary solution, I reordered the schema (lines: 110-113) by using the key present in line 111 & for the array, I did the same (lines: 106-109) by the key in line 107. By doing all of this, I expect the elements in both the array and schema to be aligned, however i don't know how robust this solution is.